src/labelle/lib/devices/device_config.py

```diff
@@ -0,0 +1,71 @@
+from __future__ import annotations
+
+import logging
+from typing import Tuple
+
+LOG = logging.getLogger(__name__)
+
+
+class DeviceConfig:
+    """Configuration object for the capabilities of a label printer."""
+
+    name: str
+    """Name of this device"""
+
+    deviceIDs: list[int]
+    """List of USB Device ID's this device can identify as"""
+
+    printHeadSizePixels: int
+    """Size of the print head in pixels (use calibration routine to determine)"""
+
+    supportedTapeSizes: list[int]
+    """List of supported tape sizes in mm"""
+
+    marginsPerTape: dict[int, Tuple[int, int]]
+    """
+    Dictonary of print margins per tape size
+    Entry: TapeSizeMM : (topMarginInPx, bottomMarginInPx)
+    Use the calibration routine to determine these for each tape size
```

Comment:
Please add a link here to a markdown document under `/doc` which describes what the calibration routine is.

Reply:
Will check

Reply:
Unfortunately there is no document at this moment. I think we need to add a link when it is there.